pkg/provider/doppler/provider.go

```diff
@@ -61,9 +89,18 @@ func (p *Provider) NewClient(ctx context.Context, store esv1.GenericStore, kube
 
 	dopplerStoreSpec := storeSpec.Provider.Doppler
 
-	// Default Key to dopplerToken if not specified
-	if dopplerStoreSpec.Auth.SecretRef.DopplerToken.Key == "" {
-		storeSpec.Provider.Doppler.Auth.SecretRef.DopplerToken.Key = "dopplerToken"
+	useCache := dopplerStoreSpec.Auth.OIDCConfig != nil
+
+	key := cache.Key{
+		Name:      store.GetObjectMeta().Name,
+		Namespace: namespace,
+		Kind:      store.GetTypeMeta().Kind,
+	}
```

Comment:
nit/optional: We should probably disambiguate any references to caches and keys by including auth/oidc somewhere in the name, in anticipation of also caching secret values.